content/ros/ROS_Nodes_Topics_Messages/README.md

```diff
@@ -0,0 +1,280 @@
+
+A package is nothing more than a folder with at least these two files:
+    
+    - package.xml
+    - CMakeLists.txt
+
+Normally, a package also includes Python scripts, C++ source files, launch files (we will see later), message definition files, service definition files, ...etc. It 
+is common for a ROS package to have the structure shown below:
+
+```
+package/
+├── CMakeLists.txt
+├── include/
+├── msg/
+├── package.xml
+├── scripts/
+├── src/
+└── srv/
+```
+ ### How?
+- Navigate to the ```src``` folder of your ROS workspace.
+```
+cd ~/catkin_ws/src/
+```
+
+This command assumes the workspace is located in the home directory and named as ```catkin_ws```. The ROS installation script which you used for installing ROS, let's you choose a name for this workspace. If you don't have a workspace for some reason, see the [installation page](https://github.com/mas-group/foundations_course/tree/master/content/ros/ROS%20Installation).
+
+- Use the ```catkin_create_pkg``` command to create the package for you. This 
+commad has the following syntax:
+
+```
+catkin_create_pkg <package_name> [depend1] [depend2] [depend3]
+```
+Where ```depend1, depend2, depend3```  are [dependencies](#dependencies) you might use in your package.  
+
+
+
+Creat a package named as ```my_first_package``` with ```rospy``` and ```std_msgs``` as dependencies:
+```
+catkin_create_pkg my_first_package std_msgs rospy 
+```
+- [Compile](#compilation) using ```catkin build``` command:
+```
+catkin build
+```
+You're done!  
+You can now navigate to this package direclty using ```roscd```:
+
+- One more thing:
+
+ROS uses the ```ROS_PACKAGE_PATH``` environment variable when it looks up for packages and nodes. Try to see where it points at:
+
+```
+echo $ROS_PACKAGE_PATH
+```
+
+</details>
+
+
+
+
+## Using ```rosrun``` to run a node
+:warning: Don't forget to run ROS master before doing any of the instructions that follows. To run the master, use the following command:
+
+```
+roscore
+```
+
+```rosrun``` can be used to run a node directly without the need to navigate to it's directory. 
+
+```
+rosrun <package> <executable> 
+```
+- create a ```scripts``` folder inside ```my_first_package/scripts``` and place our scripts there.
+- In a terminal (<kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>t</kbd>), navigate to the ```scripts``` folder.
+```
+roscd my_first_package/scripts/
+```
+
+- Give the script files execuation permission. Example:
+```
+sudo chmod +x 01_simple_node.py
+```
+:bulb: you can do that for all the files ending with ```.py``` in one command:
+```
+sudo chmod +x *.py
+```
+- use ```rosrun```:
+```
+rosrun my_first_package 01_simple_node.py
+```
+
+
+<details><summary>More details:</summary>
+
+<br/>
+
+- ```rosrun``` searches withing the directory tree of the given package. The scripts doesn't even have to be located in ```scripts``` folder. It's only common between ROS users to place Python scripts inside the ```scripts``` folder.
```

Comment:
doesn't -> don't

in `scripts` folder -> in the `scripts` folder

Reply:
done